router/core/access_controller.go

```diff
@@ -16,15 +33,29 @@ var (
 
 // AccessController handles both authentication and authorization for the Router
 type AccessController struct {
-	authenticationRequired bool
-	authenticators         []authentication.Authenticator
+	authenticationRequired     bool
+	authenticators             []authentication.Authenticator
+	introspectionAuthMode      IntrospectionAuthMode
+	introspectionAuthSkipToken string
 }
 
-func NewAccessController(authenticators []authentication.Authenticator, authenticationRequired bool) *AccessController {
-	return &AccessController{
-		authenticationRequired: authenticationRequired,
-		authenticators:         authenticators,
+// NewAccessController creates a new AccessController.
+// It returns an error if the introspection auth mode is invalid.
+func NewAccessController(
+	authenticators []authentication.Authenticator,
+	authenticationRequired bool,
+	introspectionAuthMode IntrospectionAuthMode,
+	introspectionAuthSkipToken string) (*AccessController, error) {
+	if introspectionAuthMode != IntrospectionAuthModeFull && introspectionAuthMode != IntrospectionAuthModeToken && introspectionAuthMode != IntrospectionAuthModeSkip {
+		return nil, fmt.Errorf("invalid introspection auth mode: %s", introspectionAuthMode)
 	}
```

Comment:
Total nitpick, but you could attach a validator to the type and invert this, it would be faster. As-is you always check 4 inequalities instead of =<4 equalities